File: app/src/main/java/com/example/domain/model/Account.kt
package com.example.domain.model

data class Account(
    val id: Long = 0,
    val code: String,
    val name: String,
    val parentId: Long?,
    val level: Int,
    val accountType: AccountType,
    val description: String = "",
    val isActive: Boolean = true,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    val isPostingAccount: Boolean
        get() = level == 4 && isActive

    val levelDescription: String
        get() = when (level) {
            1 -> "Level 1 · Major Head"
            2 -> "Level 2 · Sub-Head"
            3 -> "Level 3 · Control Group"
            4 -> "Level 4 · Posting Account"
            else -> "Level $level"
        }

    val displayLabel: String
        get() = "$code - $name"
}

File: backend/src/Accounting.Domain/Entities/Account.cs
using Accounting.Domain.Common;

namespace Accounting.Domain.Entities;

public class Account : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public int Level { get; set; } // 1, 2, 3, 4
    public long? ParentId { get; set; }
    public AccountType AccountType { get; set; }
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;

    // Computed rule: Only Level 4 accounts can receive journal/voucher entries
    public bool IsPostingAccount => Level == 4;

    // Navigation Properties
    public virtual Account? ParentAccount { get; set; }
    public virtual ICollection<Account> Children { get; set; } = new List<Account>();
    public virtual ICollection<VoucherLine> VoucherLines { get; set; } = new List<VoucherLine>();
    public virtual ICollection<LedgerTransaction> LedgerTransactions { get; set; } = new List<LedgerTransaction>();
}
